### PR TITLE
Create List Report

### PR DESCRIPTION
**Pull request by ilBury**

### CODE REVIEW

**Comment by ho-thach-nguyen-lx** [comment]
Please rename the PR name to something like "Create List Report". It should reflect the job you've done

File: package.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "3.-appbase",
```
Comment (ho-thach-nguyen-lx) [inline package.json]: Please rename. Every single line of code you wrote should make sense

File: package.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  "version": "1.0.0",
+  "description": "",
+  "main": "Component.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
```
Comment (ho-thach-nguyen-lx) [inline package.json]: Remove it if not used

File: ui5.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+specVersion: "3.2"
+metadata:
+  name: appbase
```
Comment (ho-thach-nguyen-lx) [inline ui5.yaml]: Please rename

File: ui5.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  customMiddleware:
+    - name: fiori-tools-proxy
+      afterMiddleware: compression
+      configuration:
+        ignoreCertError: false
+        ui5:
+          path:
+            - /resources
+            - /test-resources
+          url: https://openui5.hana.ondemand.com
```
Comment (ho-thach-nguyen-lx) [inline ui5.yaml]: It's better to use sapui5 instead of openui5

File: webapp/model/products.json [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+            "Name": "car",
+            "Description": "very comfortable thing",
+            "Price": 21.65,
+            "Quantity": "22",
+            "Supplier": ["1", "3", "5"],
+            "Category": "speed",
+            "Availability": false
+        }
+    ],
+    "Categories": [
```
Comment (ho-thach-nguyen-lx) [inline webapp/model/products.json]: I asked you to split products.json file into 3 separate ones so you could remove Categories and Suppliers entities from products.json file

File: webapp/controller/PageOverview.controller.js [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+		onInit: function() {
+			const oModel = new JSONModel();
+			oModel.loadData("../model/products.json");
+		
+			this.getView().setModel(oModel);
+		},
+
+		onAfterRendering: function() {
+			const oMultiInput = this.byId("multiInput");
+			oMultiInput.attachTokenUpdate((event) => {
```
Comment (ho-thach-nguyen-lx) [inline webapp/controller/PageOverview.controller.js]: Why not define tokenUpdate event in xml view and attach handler from controller to it?

Comment (ilBury) [reply webapp/controller/PageOverview.controller.js]: I didn't use it, because in documentation updateToken like deprecated, and I thought that it is could be a mistake 
Update: sorry, it is changeToken deprecated 
Update2: changed as you said

File: webapp/controller/PageOverview.controller.js [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+			const aSuppliers = oModel.getProperty("/Suppliers");			
+			
+			return aSuppliers
+					.filter((el) => data.includes(el.SupplierId))
+					.map(el => el.SuppliersName)
+					.join(", ");
+		},
+
+		getCombinedFilter: function(aSelectedId) {
+			const sQuerySearch = this.byId("searcher").getValue();
```
Comment (ho-thach-nguyen-lx) [inline webapp/controller/PageOverview.controller.js]: You should never get data from controls directly. You should use model instead and get data from it.

something like that in xml:
```xml
<m:SearchInput value="{ui>/querySearch}"/>
```

something like that in controller:
```js
const sQuerySearch = this.getView().getModel("ui").getProperty("/querySearch")
```

Don't forget to initialise ui model

File: webapp/controller/PageOverview.controller.js [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+	"sap/ui/model/FilterOperator",
+	'sap/m/Token',
+], function (Controller, JSONModel, Filter, FilterOperator, Token) {
+	"use strict";
+
+	return Controller.extend("products.app.controller.PageOverview", {
+		
+		onInit: function() {
+			const oModel = new JSONModel();
+			oModel.loadData("../model/products.json");
```
Comment (ho-thach-nguyen-lx) [inline webapp/controller/PageOverview.controller.js]: You should load all data from model folder, not just products.json

File: webapp/controller/PageOverview.controller.js [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+					.filter((el) => data.includes(el.SupplierId))
+					.map(el => el.SuppliersName)
+					.join(", ");
+		},
+
+		getCombinedFilter: function(aSelectedId) {
+			const sQuerySearch = this.byId("searcher").getValue();
+			const sSelectedKey = this.byId("availabilitySelect").getSelectedKey();
+			
+			const sAvailability = "all";
```
Comment (ho-thach-nguyen-lx) [inline webapp/controller/PageOverview.controller.js]: Let's make the availability filtering work in this way: when users want to see all possible products, they should either choose all ("In Stock" and "Out Of Stock") values or choose nothing, meaning the Availability filter is empty.

File: webapp/controller/PageOverview.controller.js [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+			const oAvailabilityFilter = new Filter(
+				"Availability", 
+				FilterOperator.NE,
+				sSelectedKey === sAvailability ? sSelectedKey : JSON.parse(sSelectedKey.toLowerCase())
```
Comment (ho-thach-nguyen-lx) [inline webapp/controller/PageOverview.controller.js]: 1. Do you really need to JSON.parse here?
2. sSelectedKey means the value is of type string. If it is not string, please change the notation (oSelectedKey, mSelectedKey etc). But I believe the value is of type string and you don't need any parsing here

File: webapp/model/products.json [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+            "Availability": true
+        },
+        {
+            "Id": "2",
+            "Name": "coffee",
+            "Description": "very cheerful thing",
+            "Price": 21.65,
+            "Quantity": "22",
+            "Supplier": ["1", "3"],
+            "Category": "enjoy",
```
Comment (ho-thach-nguyen-lx) [inline webapp/model/products.json]: Product's category should not be name but id, so in this line you should change it to:
```suggestion
            "Category": "2",
```

File: webapp/controller/PageOverview.controller.js
Comment (ho-thach-nguyen-lx) [inline webapp/controller/PageOverview.controller.js:1]: Please rename this file to ListReport.controller.js
It's the best practice to name the file like that in ListReport floorplan based applications

File: webapp/controller/PageOverview.controller.js [hunk 12]
```diff
@@ -0,0 +1,199 @@
...
+
+		getSuppliersFilters: function(aSelectedId) {
+			const oMultiInput = this.byId("multiInput");
+			const oModel = this.getView().getModel();
+			const aAllSuppliers = oModel.getProperty("/Suppliers");
+			const aSuppliersItems = oMultiInput.getTokens().map(el => el.getText());
+			const aFilters = [];
+			let aSuppliersId = null;
+		
+			if(aSelectedId) {
```
Comment (ho-thach-nguyen-lx) [inline webapp/controller/PageOverview.controller.js]: The condition is not quite correct. As aSelectedId is of type array, it will be always true

File: webapp/controller/PageOverview.controller.js [hunk 13]
```diff
@@ -0,0 +1,199 @@
...
+			const aFilters = aCategoryItems.map((el) => {
+				return new Filter("Category", FilterOperator.Contains, el.getText())
+			})
+			
+			return aFilters.length ? aFilters : [new Filter("Category", FilterOperator.Contains, "")];
+		},
+
+		
+
+		handleValueHelp: function(event) {
```
Comment (ho-thach-nguyen-lx) [inline webapp/controller/PageOverview.controller.js]: Please rename to handleValueHelpRequest. It's always good to have event name so we could understand what it is responsible for and to what it's bound.

File: webapp/controller/PageOverview.controller.js [hunk 14]
```diff
@@ -0,0 +1,199 @@
...
+			const oView = this.getView();
+			const oModel = oView.getModel();
+			if (!this.oDialog) {	
+				this.oDialog = sap.ui.xmlfragment(oView.getId(), "products.app.view.fragments.SuppliersDialog", this);
```
Comment (ho-thach-nguyen-lx) [inline webapp/controller/PageOverview.controller.js]: do not use sap.ui.xmlfragment as it's deprecated. Use Fragment.load instead. Find api in the docs

File: webapp/controller/PageOverview.controller.js [hunk 15]
```diff
@@ -0,0 +1,199 @@
...
+
+				oView.addDependent(this.oDialog);
+			}
+		
+			this.oDialog.setModel(oModel);
```
Comment (ho-thach-nguyen-lx) [inline webapp/controller/PageOverview.controller.js]: There's a thing in sapui5 called model propagation. Once you created a control and added it to the view 'dependents' aggregation, it has all access to all models of the view.

In your case, you created this.oDialog, then added it to oView by
```ts
oView.addDependent(this.oDialog)
```

So now this.oDialog has all access to all models of oView

File: webapp/controller/PageOverview.controller.js [hunk 16]
```diff
@@ -0,0 +1,199 @@
...
+
+			this.oDialog.open();
+		},
+
+		markSelectedFields: function() {
```
Comment (ho-thach-nguyen-lx) [inline webapp/controller/PageOverview.controller.js]: I don't see you use this function anywhere. Do we really need it?

File: webapp/controller/PageOverview.controller.js [hunk 17]
```diff
@@ -0,0 +1,199 @@
...
+					el.setSelected(true)
+				}
+			})
+
+		},
+
+		_handleValueHelpSearch: function(event) {
+			const sValue = event.getParameter("value");
+			const oFilter = new Filter(
+				"SuppliersName",
```
Comment (ho-thach-nguyen-lx) [inline webapp/controller/PageOverview.controller.js]: Please filter by address as well

File: webapp/index.html [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+<html lang="en">
+<head>
+	<meta http-equiv="X-UA-Compatible" content="IE=edge" />
+	<meta http-equiv="Content-Type" content="text/html;charset=UTF-8"/>
+	<meta name="viewport" content="width=device-width, initial-scale=1.0" />
+	<title>List Report</title>
+
+	<!-- Bootstrap the UI5 core library -->
+	<script id="sap-ui-bootstrap"
+			src="https://sapui5.netweaver.ondemand.com/resources/sap-ui-core.js"
```
Comment (ho-thach-nguyen-lx) [inline webapp/index.html]: where did you get this link? https://sapui5.hana.ondemand.com is usually used

File: webapp/index.html [hunk 19]
```diff
@@ -0,0 +1,29 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+	<meta http-equiv="X-UA-Compatible" content="IE=edge" />
+	<meta http-equiv="Content-Type" content="text/html;charset=UTF-8"/>
+	<meta name="viewport" content="width=device-width, initial-scale=1.0" />
+	<title>List Report</title>
```
Comment (ho-thach-nguyen-lx) [inline webapp/index.html]: The app is not about List Report, it's about Products. List Report is a name of one of the pages

File: webapp/model/suppliers.json [hunk 20]
```diff
@@ -0,0 +1,44 @@
+{
+    "Suppliers": [
+        {
+            "SupplierId": "1",
+            "SuppliersName": "Gippo",
+            "Address": "asdsad",
```
Comment (ho-thach-nguyen-lx) [inline webapp/model/suppliers.json]: Please give meaningful address. Not random letters

File: webapp/view/PageOverview.view.xml [hunk 21]
```diff
@@ -0,0 +1,203 @@
+<mvc:View
+	xmlns="sap.m"
+	xmlns:mvc="sap.ui.core.mvc"
+	xmlns:f="sap.f"
+	xmlns:core="sap.ui.core"
+	xmlns:fb="sap.ui.comp.filterbar"
+	xmlns:layout="sap.ui.layout"
```
Comment (ho-thach-nguyen-lx) [inline webapp/view/PageOverview.view.xml]: I don't see you use this dependency anywhere. Do we really need it?

File: webapp/view/PageOverview.view.xml [hunk 22]
```diff
@@ -0,0 +1,203 @@
...
+	xmlns:svm="sap.ui.comp.smartvariants"
```
Comment (ho-thach-nguyen-lx) [inline webapp/view/PageOverview.view.xml]: I don't see you use this dependency anywhere. Do we really need it?

File: webapp/view/PageOverview.view.xml [hunk 23]
```diff
@@ -0,0 +1,203 @@
...
+				<f:DynamicPageHeader>
+					<f:content>
+						
+						<fb:FilterBar id="filterbar" persistencyKey="myPersKey" useToolbar="false" >
+							<fb:filterGroupItems >
+
+								<fb:FilterGroupItem 
+									name="search" label="Product name" 
+									groupTitle="false" 
+									groupName="Group1"
```
Comment (ho-thach-nguyen-lx) [inline webapp/view/PageOverview.view.xml]: Please provide a meaningful group name

File: webapp/view/PageOverview.view.xml [hunk 24]
```diff
@@ -0,0 +1,203 @@
...
+		<f:DynamicPage 
+			id="dynamicPageId" 
+			preserveHeaderStateOnScroll="true" 
+			showFooter="true" 
+			fitContent="true">
+			<!-- DynamicPage Title -->
+			<f:title>
+				<f:DynamicPageTitle>
+					<f:heading>
+						<Title text="Standart"/>
```
Comment (ho-thach-nguyen-lx) [inline webapp/view/PageOverview.view.xml]: This is out of scope of requirements. Please remove it

File: webapp/view/PageOverview.view.xml [hunk 25]
```diff
@@ -0,0 +1,203 @@
...
+					</f:heading>
+					
+					<f:actions>
+						<ToolbarSpacer/>
+						
+						<Button
```
Comment (ho-thach-nguyen-lx) [inline webapp/view/PageOverview.view.xml]: This is out of scope of requirements. Please remove it

**Review by ho-thach-nguyen-lx — CHANGES_REQUESTED** [review]
Hi, I am quite impressed by your work. In general your code is quite clean, easy to read and understand. Good job!

But there are several things you need to work on:
1. Answer the all questions 
2. Adjust code according to my comments
3. Add i18n. Your application could be used by people from different countries who speak different languages, so we need to make the app accessible for them.
4. Finalise price range filter
5. Fix bugs (I will tell about later and you can find them in the comments below)

**Comment by ho-thach-nguyen-lx** [comment]
Bug 1
I try to apply "Supplier name" filter. 
1. I open Supplier name value help
2. Choose "Rud"
3. Close value help
4. Open value help again

Expected behavior: Rud is selected
Actual behavior: Rud is not selected

Screenshot 2024-02-27 alle 10 18 17

**Comment by ho-thach-nguyen-lx** [comment]
When I choose all suppliers, I see the empty table. Instead I should see all data
Screenshot 2024-02-27 alle 10 22 14

**Comment by ho-thach-nguyen-lx** [comment]
Filter by description is missing
Currency is missing

File: webapp/view/PageOverview.view.xml (hunk 22 above)
Comment (ilBury) [reply webapp/view/PageOverview.view.xml]: No, was removed

File: webapp/view/PageOverview.view.xml (hunk 21 above)
Comment (ilBury) [reply webapp/view/PageOverview.view.xml]: No, was removed

File: webapp/index.html (hunk 18 above)
Comment (ilBury) [reply webapp/index.html]: Ok, was changed

File: webapp/controller/PageOverview.controller.js (hunk 16 above)
Comment (ilBury) [reply webapp/controller/PageOverview.controller.js]: No, forgot delete this function, was removed

File: webapp/controller/PageOverview.controller.js (hunk 10 above)
Comment (ilBury) [reply webapp/controller/PageOverview.controller.js]: I need JSON.parse(sSelectedKey.toLowerCase()) because it help me parse string like "false" and "true" in boolean false and true. I just found this in internet). Better to replace this?

Comment (ho-thach-nguyen-lx) [reply webapp/controller/PageOverview.controller.js]: > I need JSON.parse(sSelectedKey.toLowerCase()) because it help me parse string like "false" and "true" in boolean false and true. I just found this in internet). Better to replace this?

Okay. Then leave it like that

File: webapp/controller/ListReport.controller.js [hunk 26]
```diff
@@ -0,0 +1,302 @@
...
+				]
+			})
+
+			const sProductsJson = "../model/products.json";
+			const sCategoriesJson = "../model/categories.json";
+			const sSuppliersJson = "../model/suppliers.json";
+
+			oModelProducts.loadData(sProductsJson, "", false);
+			oModelCategories.loadData(sCategoriesJson, "", false);
+			oModelSuppliers.loadData(sSuppliersJson, "", false);
```
Comment (ho-thach-nguyen-lx) [inline webapp/controller/ListReport.controller.js]: I'd like to give an advice - always check the API in the docs so you can use it to the max. Here you created 3 json models, then load data to them, then change the data. Quite complicated. Here is an example how you could do it easily
```ts
const oModel = new JSONModel();

oModel.loadData("../model/products.json");
oModel.loadData("../model/categories.json");
oModel.loadData("../model/suppliers.json");
```

#loadData loads data to the model and in your case nothing will be overwrited because each json file has different data structure

File: webapp/controller/ListReport.controller.js [hunk 27]
```diff
@@ -0,0 +1,302 @@
...
+
+			oModelCategories.setData(oModelSuppliers.getData(), true);
+			oModelProducts.setData(oModelCategories.getData(), true);
+			
+			this.getView().setModel(oModelProducts);
+			this.getView().setModel(oPriceRangeModel, "PriceModel")
+		},
+
+		getPriceFilter: function() {
+			const sSelectedKey = this.byId("priceSelect").getSelectedKey();
```
Comment (ho-thach-nguyen-lx) [inline webapp/controller/ListReport.controller.js]: As I mentioned earlier, no data should be retrieved from control. 
Bind model to control (view), then in controller retrieve data from model.

This is MVC. 

In controller you should not interact with controls a lot

File: webapp/controller/ListReport.controller.js [hunk 28]
```diff
@@ -0,0 +1,302 @@
...
+				});
+				
+				const aSelectedTokens = aRemainingTokens.map(el => el.getText());
+				const aSelectedId = aSuppliers
+										.filter((el) => aSelectedTokens.includes(el.SuppliersName))
+										.map((el) => el.SupplierId);
+				
+				this.onFilter(aSelectedId);
+				
+			} else {
```
Comment (ho-thach-nguyen-lx) [inline webapp/controller/ListReport.controller.js]: Please remove empty block or provide some logic if token wasn't removed

File: webapp/controller/ListReport.controller.js [hunk 29]
```diff
@@ -0,0 +1,302 @@
...
+					this.oDialog = oDialog
+					oView.addDependent(this.oDialog);
+					
+					this.oDialog.open();
+				})
+			} else {
+				const oMultiInput = this.byId("multiInput");
+				const aDialogList = this.byId("valueHelpDialog");
+				const aSelectedTokens = oMultiInput.getTokens().map(el => el.getText());
+				aDialogList.getItems().map(el => el.setSelected(false));
```
Comment (ho-thach-nguyen-lx) [inline webapp/controller/ListReport.controller.js]: This code works but it neglects MVC principles.
You should not change the state of controls in controller. The state of control should be controlled by data (model).

This is very important because if you always change control's state in controller, your code will grow very fast, it is going to be difficult to debug, refactor, read, maintain and change the code base.

Please keep tokens data in model and use them in view.

something like that:

```xml
<StandardListItem
			...

			selected="{= ${ui/tokenData}.includes(${Id})}"

			... />
```

File: webapp/i18n/i18n.properties [hunk 30]
```diff
@@ -0,0 +1,13 @@
+ListReportTitle=Products
+ProductText=Product
+NameText=Name
```
Comment (ho-thach-nguyen-lx) [inline webapp/i18n/i18n.properties]: Name it ProductName because you are going more names in the app.  Product Name, Supplier Name, Category Name and so on.

My advice is to use the following pattern:

Data entity -> Its property -> Control
Product           Name                Text

So the result is
ProductNameText

Comment (ho-thach-nguyen-lx) [reply webapp/i18n/i18n.properties]: Please review all i18n keys

File: webapp/view/ListReport.view.xml [hunk 31]
```diff
@@ -0,0 +1,187 @@
...
+				<f:DynamicPageHeader>
+					
+					<f:content>
+						
+						<fb:FilterBar id="filterbar" persistencyKey="myPersKey" useToolbar="false" >
+							<fb:filterGroupItems >
+
+								<fb:FilterGroupItem 
+									name="search" 
+									label="{i18n>NameText} {i18n>AndText} {i18n>DescriptionText}"
```
Comment (ho-thach-nguyen-lx) [inline webapp/view/ListReport.view.xml]: This is not quite great because people from Western Asia could use your application. They read right to left, not left to right like most of the world. So you will need to use placeholders and templates of i18n. 

Tomorrow I will show you how to do it.

Comment (ho-thach-nguyen-lx) [reply webapp/view/ListReport.view.xml]: Placeholder will change app from 'left to right' to 'right to left' so people from Western Asia could use your apps

File: webapp/view/ListReport.view.xml [hunk 32]
```diff
@@ -0,0 +1,187 @@
...
+					
+					<f:content>
+						
+						<fb:FilterBar id="filterbar" persistencyKey="myPersKey" useToolbar="false" >
+							<fb:filterGroupItems >
+
+								<fb:FilterGroupItem 
+									name="search" 
+									label="{i18n>NameText} {i18n>AndText} {i18n>DescriptionText}" 
+									groupTitle="false"
```
Comment (ho-thach-nguyen-lx) [inline webapp/view/ListReport.view.xml]: Why title is false?

File: webapp/view/ListReport.view.xml [hunk 33]
```diff
@@ -0,0 +1,187 @@
...
+					<f:content>
+						
+						<fb:FilterBar id="filterbar" persistencyKey="myPersKey" useToolbar="false" >
+							<fb:filterGroupItems >
+
+								<fb:FilterGroupItem 
+									name="search" 
+									label="{i18n>NameText} {i18n>AndText} {i18n>DescriptionText}" 
+									groupTitle="false" 
+									groupName="HeaderFiltersGroup"
```
Comment (ho-thach-nguyen-lx) [inline webapp/view/ListReport.view.xml]: Always think about how people could understand your code. Header Filters Group is good, it shows that a field belongs to the filtering group and it can change how data appears. But you also could provide more details, by renaming it to ProductFilterGroup. So it would be easier to understand its domain, I think

File: webapp/view/ListReport.view.xml [hunk 34]
```diff
@@ -0,0 +1,187 @@
...
+									groupName="HeaderFiltersGroup" 
+									visibleInFilterBar="true">
+									<fb:control>
+										<Select
+											id="availabilitySelect"
+											forceSelection="false"
+											selectedKey="all"
+											change="onFilter(null)"
+										>
+										<core:Item key="all" text="all" />
```
Comment (ho-thach-nguyen-lx) [inline webapp/view/ListReport.view.xml]: Please remove all. Let user select all or nothing in order to see all products in the table

File: webapp/view/ListReport.view.xml [hunk 35]
```diff
@@ -0,0 +1,187 @@
...
+										<core:Item key="false" text="In Stock" />
+										<core:Item key="true" text="Out of Stock" />
+									</Select>
+									</fb:control>
+								</fb:FilterGroupItem>
+
+								<fb:FilterGroupItem 
+									name="Price range"
```
Comment (ho-thach-nguyen-lx) [inline webapp/view/ListReport.view.xml]: No hardcoded text.
You could possibly think that it is not displayed in the UI so it doesn't need i18n. But this is partially true.
Your application can be used by people who have vision disorders, so they will use tools like screen readers which will tell them what they are interacting with.

In example, a blind from Germany uses your app. He wants to filter products by price, but what he hears is "Price range", not german "Preisklasse". He confused.

Please use i18n here and everywhere else

File: webapp/view/ListReport.view.xml [hunk 36]
```diff
@@ -0,0 +1,187 @@
...
+										path: 'Supplier',
+										formatter: '.getSuppliersName'
+									}" 
+								/>
+								
+								<Text text="{
+									path: 'Category',
+									formatter: '.getCategoriesName'
+								}" />
+								<Text text="{= ${Availability} === true ? 'In Stock' : 'Out of Stock'}" />
```
Comment (ho-thach-nguyen-lx) [inline webapp/view/ListReport.view.xml]: No hardcoded text

**Review by ho-thach-nguyen-lx — CHANGES_REQUESTED** [review]
Looks much better!
But there are some things improve.
Please have a look at the comments below. Some things I will explain tomorrow. Or you can find the info by yourself which would be great

**Comment by ho-thach-nguyen-lx** [comment]
Trim the query
Screenshot 2024-02-28 alle 16 35 58

**Comment by ho-thach-nguyen-lx** [comment]
> When I choose all suppliers, I see the empty table. Instead I should see all data Screenshot 2024-02-27 alle 10 22 14

This is not fixed

**Comment by ho-thach-nguyen-lx** [comment]
Please remove unused variables

File: webapp/view/ListReport.view.xml (hunk 34 above)
Comment (ilBury) [reply webapp/view/ListReport.view.xml]: I don't understand a bit, I just should delete text="all" or I could leave it, but with i18n?

Comment (ilBury) [reply webapp/view/ListReport.view.xml]: Or you mean that user could press on empty field and in table will be all products?

Comment (ho-thach-nguyen-lx) [reply webapp/view/ListReport.view.xml]: > I don't understand a bit, I just should delete text="all" or I could leave it, but with i18n?

Yes, remove 'all'. There should be only 2 items in select control. If select is empty, show all data in table, if all items are selected, show data in table as well.

File: webapp/view/ListReport.view.xml (hunk 35 above)
Comment (ilBury) [reply webapp/view/ListReport.view.xml]: But we can't to add in "name" text from i18n like this: name="{i18n>PriceRangeText}" , or with parts and formatter. It doesn't work for property name

Comment (ho-thach-nguyen-lx) [reply webapp/view/ListReport.view.xml]: Interesting. Will check. For now just leave it

File: webapp/controller/ListReport.controller.js (hunk 26 above)
Comment (ilBury) [reply webapp/controller/ListReport.controller.js]: It doesn't work, when I try to get data, returns always undefined. That's why I made it so complicated.
And only the last json is written, in this case only suppliers.

File: webapp/view/ListReport.view.xml [hunk 37]
```diff
@@ -0,0 +1,221 @@
...
+		<f:DynamicPage 
+			id="dynamicPageId" 
+			preserveHeaderStateOnScroll="true" 
+			showFooter="true" 
+			fitContent="true">
+			<!-- DynamicPage Title -->
+			<f:title>
+				<f:DynamicPageTitle>
+					<f:heading>
+						<!-- <Title text="{i18n>ListReportTitle}"/> -->
```
Comment (ho-thach-nguyen-lx) [inline webapp/view/ListReport.view.xml]: Please remove useless comments

**Review by ho-thach-nguyen-lx — APPROVED** [review]
As discussed, all good. There are some things to improve but we'll do it lately
